projects/packages/publicize/src/class-publicize-script-data.php

```diff
@@ -221,9 +223,30 @@ public static function get_shares_data() {
 	/**
 	 * Get the list of supported Publicize services.
 	 *
+	 * @param bool $force_refresh Whether to force a refresh of the services list.
 	 * @return array List of external services and their settings.
 	 */
-	public static function get_supported_services() {
+	public static function get_supported_services( $force_refresh = false ) {
+		if ( defined( 'IS_WPCOM' ) && constant( 'IS_WPCOM' ) ) {
+			if ( function_exists( 'require_lib' ) ) {
+				// @phan-suppress-next-line PhanUndeclaredFunction - phan is dumb not to see the function_exists check.
+				require_lib( 'external-connections' );
+			}
+
+			// @phan-suppress-next-line PhanUndeclaredClassMethod - We are here because we are on WPCOM.
+			$external_connections = \WPCOM_External_Connections::init();
+			$services             = array_values( $external_connections->get_external_services_list( 'publicize', get_current_blog_id() ) );
+
+			return $services;
+		}
+
+		// Checking the cache.
+		$services = get_transient( self::SERVICES_TRANSIENT );
+		if ( false !== $services && ! $force_refresh ) {
+			return $services;
+		}
+
+		// Fetch the services.
```

Comment:
@pablinos I feel like these small things keep building up that result in what our publicize classes are right now. Is it worth having this extra logic in a services class like [this](https://github.com/Automattic/jetpack/pull/40536/files#diff-b63c86018aa6f501ff45fd519d9b9ee414f31e8971c1de05bfb4d0cdb88591d0) to keep this script data class cleaner?

Reply:
Yes it would probably be worth doing that if there's a group of Services related functions that could live together. Otherwise, we need to watch for the tipping point where things get out of control, but I lean towards having things in the same place, rather than lots of files with one function in them.

Reply:
Currently it's only this function in this file, but I'm happy to create a new class for it if you want

Reply:
Yeah, when we refactor those publicize classes, we can move more of that services related logic to that class.

Reply:
Done, have to wait for the build to test on simple and atomic